File: rust/axum/src/modules/branch/controller.rs
use axum::{
    response::IntoResponse,
    Json,
};

use super::service::BranchService;

pub struct BranchController{
}


impl BranchController {
    pub async fn find_all() -> impl IntoResponse {
        match BranchService::find_all().await {
            Ok(branches) => Json(branches).into_response(),
            Err(err) => err.into_response(),
        }
    }
}

File: go/fiber_orm/internal/modules/branch/controller.go
package branch

import (
	"github.com/gofiber/fiber/v2"
)

type BranchController struct{
	service BranchService
}

func (c *BranchController) FindAll(ctx *fiber.Ctx) error {
	branchs, err := c.service.FindAll()

	if err != nil {
		return ctx.Status(err.StatusCode).JSON(err)
	}

	return ctx.Status(fiber.StatusOK).JSON(branchs)
}